### PR TITLE
[InHouse] Allow init w/o adUnit

### PR DESCRIPTION
**Pull request by vguerci**
As it is redundant with loadWithBid. As a consequence, remove check
that adUnit is the one that was used at init

JIRA: EE-1332

### CODE REVIEW

File: CriteoPublisherSdk/Sources/Native/CRNativeLoader.m [hunk 1]
```diff
@@ -41,6 +41,13 @@ @implementation CRNativeLoader
 
 #pragma mark - Life cycle
 
+- (instancetype)init {
```
Comment (TurpIF) [inline CriteoPublisherSdk/Sources/Native/CRNativeLoader.m:44]: Using this init does not initialize `self.criteo` or `self.urlOpener`. Is it intended ?
Generally when overloading constructors, constructor will less parameters calls the one with more parameters with default value or nil (nil here). So the object is always initialized in the same manner.

**Review by TurpIF — CHANGES_REQUESTED** [review]
What do you think about updating the InHouse functional tests to use the constructors without ad unit ?
So this constructor would be under test.

File: CriteoPublisherSdk/Sources/Native/CRNativeLoader.m (hunk 1 above)
Comment (vguerci) [reply CriteoPublisherSdk/Sources/Native/CRNativeLoader.m:44]: Done.

**Comment by vguerci** [comment]
> What do you think about updating the InHouse functional tests to use the constructors without ad unit ?
> So this constructor would be under test.

This functional test use a shared utility checker that does the init, so it wasn't really possible. Instead added some unit tests on the class test suites which effectively tests `loadAdWithBid:` w/o an adUnit.

**Comment by vguerci** [comment]
Also added some checks to prevent `loadAd` w/o and adUnit on standalone.

**Review by TurpIF — APPROVED** [review]

**Comment by vguerci** [comment]
Thanks!

**Pull request merged**